packages/components/src/components/hds/pagination/compact/index.ts

```diff
@@ -7,12 +7,37 @@ import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
 import { assert } from '@ember/debug';
+import type { HdsPaginationDirections, HdsPaginationRouting } from '../types';
+
+interface HdsPaginationCompactSignature {
+  Args: {
+    ariaLabel?: string;
+    showLabels?: boolean;
+    isDisabledPrev?: boolean;
+    isDisabledNext?: boolean;
+    showSizeSelector?: boolean;
+    sizeSelectorLabel?: string;
+    pageSizes?: number[];
+    currentPageSize?: number;
+    route?: string;
+    model?: unknown;
+    models?: unknown[];
+    replace?: boolean;
+    queryFunction?: (
+      page: HdsPaginationDirections,
+      pageSize?: number
+    ) => Record<string, unknown>;
+    onPageChange?: (page: HdsPaginationDirections) => void;
+    onPageSizeChange?: (pageSize: number) => void;
+  };
+  Element: HTMLDivElement;
+}
```

Comment:
If we look at the template, most of these arguments are actually coming from subcomponents. This means we can write it as such to avoid duplicating argument types:
```suggestion
interface HdsPaginationCompactSignature {
  HdsPaginationControlArrowSignature['Args'] & HdsPaginationSizeSelectorSignature['Args'] & { [...] }  
  Element: HTMLDivElement;
}
```

Reply:
I'm not sure I understand how this works in practice. `HdsPaginationControlArrowSignature['Args']` shares some, but not all of the `Args` of the pagination component, which is its parent. We might have to pair on Monday if I'm missing something.

Reply:
I was hoping for something like this (and I understand that in some instances, such as the size selector, it may not be worthy) so we avoid declaring the types twice.
```suggestion
interface HdsPaginationCompactSignature {
  Args: Omit<HdsPaginationControlArrowSignature['Args'], 'direction'> & {
    ariaLabel?: string;
    showLabels?: boolean;
    isDisabledPrev?: boolean;
    isDisabledNext?: boolean;
    showSizeSelector?: boolean;
    sizeSelectorLabel?: string;
    pageSizes?: number[];
    currentPageSize?: number;
    queryFunction?: (
      page: HdsPaginationDirections,
      pageSize?: number
    ) => Record<string, unknown>;
    onPageChange?: (page: HdsPaginationDirections) => void;
    onPageSizeChange?: (pageSize: number) => void;
  };
  Element: HTMLDivElement;
}
```

Let me know if it doesn't make sense and I'll book us some time to go through it